cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -5474,10 +5472,9 @@ TEST(Substrait, MixedSort) {
   ConversionOptions conversion_options;
   conversion_options.named_table_provider = std::move(table_provider);
 
-  ASSERT_THAT(
-      DeserializePlan(*buf, /*registry=*/nullptr, /*ext_set_out=*/nullptr,
-                      conversion_options),
-      Raises(StatusCode::NotImplemented, testing::HasSubstr("mixed null placement")));
+  ASSERT_OK_AND_ASSIGN(
+      auto plan_info, DeserializePlan(*buf, /*registry=*/nullptr, /*ext_set_out=*/nullptr,
+                                      conversion_options));
```

Comment:
Please also assert that the SortKeys have correct null_placement

Reply:
It seems that I can't get this. Here I added an assertion about the final result.

Reply:
```c++
  auto& order_by_options =
      checked_cast<const acero::OrderByNodeOptions&>(*plan_info.root.declaration.options);
  EXPECT_THAT(order_by_options.ordering.sort_keys(), ElementsAre());
```

Reply:
ok